rpc/core/blocks.go

```diff
@@ -318,7 +318,8 @@ func validateDataCommitmentRange(start uint64, end uint64) error {
 	if start >= end {
 		return fmt.Errorf("last block is smaller than first block")
 	}
-	if end > uint64(env.BlockStore.Height()) {
+	// the data commitment range is end exclusive
+	if end > uint64(env.BlockStore.Height())+1 {
 		return fmt.Errorf(
 			"last block %d is higher than current chain height %d",
```

Comment:
[not blocking] this error message seems a bit confusing b/c now it's possible for the "last" block to be higher than the current chain height. "last" can be exactly one higher than the current chain height in which case the query result should include all the blocks including the most recent one. "last" can't be two higher than the current chain height because that means the query includes a block that doesn't exist yet.

The error message could be updated to say something like: 

```go
"end height %d is higher than current chain height + 1 %d so query includes a block that doesn't exist yet" 
```
But I realize that's a mouthful so open to alternatives.

[nit] it would be nice if this function body consistently referred to one term "end" or "last" rather than mixing usage. Since other code in this PR appears to use "end", preference for that.

Reply:
https://github.com/celestiaorg/celestia-core/pull/1058/files#r1289166247 ?

Reply:
https://github.com/celestiaorg/celestia-core/issues/1059